examples/graphene_apps/python_worker/avalon_worker/receive_request.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/python3
+
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import sys
+import zmq
+import logging
+import socket
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+logger.addHandler(logging.StreamHandler(sys.stdout))
+
+# -------------------------------------------------------------------------
+
+
+class ZmqSocket():
+    """
+    ZMQ socker to receive Work Order Request and send Response.
+    """
+
+# -------------------------------------------------------------------------
+
+    def __init__(self, zmq_url, wo_processor):
+        """
+        Constructor for ZmqSocket.
+        """
+        self.wo_processor = wo_processor
+        self.zmq_url = zmq_url
+
+# -------------------------------------------------------------------------
+
+    def start_zmq_listener(self):
+        """
+        This function binds to the port configured for zmq and
```

Comment:
Is it synchronous always? meaning it always uses request over ZMQ? can't this read from the work order request?

Reply:
Python worker will wait till it gets a request from client, processes the request and then wait for next request. For now processing is synchronous.